### ISSUE TITLE
Material Table row-index always undefined

### ISSUE THREAD

**Issue by woteska** [angular/material2#12793]
#### Bug, feature request, or proposal:

**BUG**

#### What is the expected behavior?

I can get the index of a row.

#### What is the current behavior?

Indexes are always undefined.

#### What are the steps to reproduce?

https://angular-material2-issue-xxupf5.stackblitz.io

#### What is the use-case or motivation for changing an existing behavior?

Highlighting a row ; Adding drag-and-drop feature to my project; etc.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

I only tried the latest.

  "dependencies": {
    "@angular/animations": "^6.1.3",
    "@angular/cdk": "^6.4.6",
    "@angular/common": "^6.1.3",
    "@angular/compiler": "^6.1.3",
    "@angular/core": "^6.1.3",
    "@angular/forms": "^6.1.3",
    "@angular/http": "^6.1.3",
    "@angular/material": "^6.4.6",
    "@angular/platform-browser": "^6.1.3",
    "@angular/platform-browser-dynamic": "^6.1.3",
    "@angular/router": "^6.1.3",
    "core-js": "^2.5.7",
    "hammerjs": "^2.0.8",
    "rxjs": "^6.2.2",
    "zone.js": "~0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.7.4",
    "@angular/cli": "~6.1.4",
    "@angular/compiler-cli": "^6.1.3",
    "@angular/language-service": "^6.1.3",
    "@types/jasmine": "~2.8.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~10.7.1",
    "codelyzer": "~4.4.4",
    "jasmine-core": "~3.2.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~3.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~1.1.2",
    "karma-jasmine-html-reporter": "^1.3.1",
    "npm-upgrade": "^1.4.1",
    "protractor": "~5.4.0",
    "sort-package-json": "^1.15.0",
    "ts-node": "~7.0.1",
    "tslint": "~5.11.0",
    "typescript": "~2.9.2"
  }

#### Is there anything else we should know?

Angular Material Basic Table works perfectly: https://angular-acdxje-ejdusk.stackblitz.io

**Comment by geromegrignon** [comment]
Change "let i = index" for "let i = dataIndex"
It works.

> /**
>  * Context provided to the row cells when `multiTemplateDataRows` is true. This context is the same
>  * as CdkCellOutletRowContext except that the single `index` value is replaced by `dataIndex` and
>  * `renderIndex`.
>  */ 

[source](https://github.com/angular/material2/blob/34a7e382144bb63965589002d0ed6bb046ac6054/src/cdk/table/row.ts)

**Comment by woteska** [comment]
Thanks buddy, it works!

I would suggest to complete the documentation (https://material.angular.io/components/table/api) with this very very important information if it is possible.

Apart from me issue can be closed. Thanks again!

**Comment by mactechlabs** [comment]
Thanks Man. Works like a charm.

**Comment by bharath0** [comment]
Thank You Sir It is working

**Comment by ssarir** [comment]
> @geromegrignon
> 
> Thanks buddy, it works!
> 
> I would suggest to complete the documentation (https://material.angular.io/components/table/api) with this very very important information if it is possible. It is missing of it and I found many users on forums who needs the index of the rows.
> 
> Apart from me issue can be closed. Thanks again!

Thanks it also saved my day

**Comment by Wilt** [comment]
Documentation still not updated?

**Comment by Akshar11** [comment]
Please update the documentation .

**Comment by richardwfff** [comment]
I agree it would be nice to have this information in the documentation since people can spend a lot of time debugging without knowing this is the property causing the problem and not having a way to search for the issue (me included)

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
